src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,49 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+
 public class SalaryInfo {
+    private static final int INIT_SALARY_AMOUNT = 0;
+    private static final String DATA_SEPARATOR = " ";
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int NUMBER_OF_DAYS_FOR_BORDER_SHIFT = 1;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        Map<String, Integer> nameSalaryMap = fillMapWithKeys(names);
```

Comment:
That's really cool you've managed to use `Map` data structure here! However, it is strongly recommended not to use `HashMap` or two-dimensional arrays during this task.

[Code recommendations](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info#dont-use-two-dimension-arrays-and-hashmapor-any-other-map)

Reply:
Fixed, one-dimensional array for now. Screwed up with common mistakes, sry.